flyteplugins/go/tasks/plugins/k8s/dask/dask.go

```diff
@@ -293,40 +293,48 @@ func (p daskResourceHandler) GetTaskPhase(ctx context.Context, pluginContext k8s
 
 	// There is a short period between the `DaskJob` resource being created and `Status.JobStatus` being set by the `dask-operator`.
 	// In that period, the `JobStatus` will be an empty string. We're treating this as Initializing/Queuing.
-	isQueued := status == "" ||
-		status == daskAPI.DaskJobCreated ||
-		status == daskAPI.DaskJobClusterCreated
-
-	if !isQueued {
-		taskExecID := pluginContext.TaskExecutionMetadata().GetTaskExecutionID()
-		o, err := logPlugin.GetTaskLogs(
-			tasklog.Input{
-				Namespace:       job.ObjectMeta.Namespace,
-				PodName:         job.Status.JobRunnerPodName,
-				LogName:         "(User logs)",
-				TaskExecutionID: taskExecID,
-			},
-		)
-		if err != nil {
-			return pluginsCore.PhaseInfoUndefined, err
-		}
-		info.Logs = o.TaskLogs
+	// isQueued := status == "" ||
+	// 	status == daskAPI.DaskJobCreated ||
+	// 	status == daskAPI.DaskJobClusterCreated
+
+	taskExecID := pluginContext.TaskExecutionMetadata().GetTaskExecutionID()
+	o, err := logPlugin.GetTaskLogs(
+		tasklog.Input{
+			Namespace:       job.ObjectMeta.Namespace,
+			PodName:         job.Status.JobRunnerPodName,
+			LogName:         "(User logs)",
+			TaskExecutionID: taskExecID,
+		},
+	)
+	if err != nil {
+		return pluginsCore.PhaseInfoUndefined, err
 	}
+	info.Logs = o.TaskLogs
+
+	var phaseInfo pluginsCore.PhaseInfo
 
 	switch status {
 	case "":
-		return pluginsCore.PhaseInfoInitializing(occurredAt, pluginsCore.DefaultPhaseVersion, "unknown", &info), nil
+		phaseInfo = pluginsCore.PhaseInfoInitializing(occurredAt, pluginsCore.DefaultPhaseVersion, "unknown", &info)
 	case daskAPI.DaskJobCreated:
-		return pluginsCore.PhaseInfoInitializing(occurredAt, pluginsCore.DefaultPhaseVersion, "job created", &info), nil
+		phaseInfo = pluginsCore.PhaseInfoInitializing(occurredAt, pluginsCore.DefaultPhaseVersion, "job created", &info)
 	case daskAPI.DaskJobClusterCreated:
-		return pluginsCore.PhaseInfoInitializing(occurredAt, pluginsCore.DefaultPhaseVersion, "cluster created", &info), nil
+		phaseInfo = pluginsCore.PhaseInfoInitializing(occurredAt, pluginsCore.DefaultPhaseVersion, "cluster created", &info)
 	case daskAPI.DaskJobFailed:
 		reason := "Dask Job failed"
-		return pluginsCore.PhaseInfoRetryableFailure(errors.DownstreamSystemError, reason, &info), nil
+		phaseInfo = pluginsCore.PhaseInfoRetryableFailure(errors.DownstreamSystemError, reason, &info)
 	case daskAPI.DaskJobSuccessful:
-		return pluginsCore.PhaseInfoSuccess(&info), nil
+		phaseInfo = pluginsCore.PhaseInfoSuccess(&info)
+	default:
+		phaseInfo = pluginsCore.PhaseInfoRunning(pluginsCore.DefaultPhaseVersion, &info)
 	}
-	return pluginsCore.PhaseInfoRunning(pluginsCore.DefaultPhaseVersion, &info), nil
+
+	phaseVersionUpdateErr := k8s.MaybeUpdatePhaseVersionFromPluginContext(&phaseInfo, &pluginContext)
+	if phaseVersionUpdateErr != nil {
+		return phaseInfo, phaseVersionUpdateErr
+	}
```

Comment:
So far, none of the k8s plugins apart from pod have checked whether the reason changed within the same phase. This means that per task phase only a single event was sent. We need to send subsequent events to admin if the reason changes as the first event doesn't contain the log links.

As a nice side effect, the status updates at the top right of flyte console now show a lot more updates within a single phase.

At the example of the tensorflow plugin/queueing phase:

Master:

```
4/7/2024 5:19:38 PM UTC task submitted to K8s
```

Branch:

```
4/7/2024 5:30:12 PM UTC task submitted to K8s

4/7/2024 5:30:13 PM UTC JobCreated

4/7/2024 5:30:12 PM UTC Restart policy in pod template will be overwritten by restart policy in replica spec

4/7/2024 5:30:13 PM UTC Created pod: f76767053e7b04e95aef-n0-0-chief-0

4/7/2024 5:30:13 PM UTC Created service: f76767053e7b04e95aef-n0-0-chief-0

4/7/2024 5:30:13 PM UTC Created pod: f76767053e7b04e95aef-n0-0-ps-0

4/7/2024 5:30:13 PM UTC Created service: f76767053e7b04e95aef-n0-0-ps-0
```